src/reconnection.ts

```diff
@@ -49,6 +49,7 @@ export class ReconnectManager<PeerMetadata, TrackMetadata> {
   private reconnectTimeoutId: NodeJS.Timeout | null = null;
   private reconnectFailedNotificationSend: boolean = false;
   private ongoingReconnection: boolean = false;
+  private reconnectionStartedNotificationSend: boolean = false;
```

Comment:
```suggestion
  private reconnectionStartedNotificationSent: boolean = false;
```

Reply:
Also there are now three boolean variables like this: `reconnectFailedNotificationSend` and `reconnectionStartedNotificationSend` and `ongoingReconnection`
I wonder if we can merge them into one variable that keeps the current state of reconnection?

Reply:
I've removed these variables and introduced a simple state machine
``` 
private status: 'INITIAL' | 'RECONNECTING' | 'FAILED' = 'INITIAL';
```